src/components/Dashboard/MainInput/MainInput.jsx

```diff
@@ -41,9 +42,12 @@ export default class MainInput extends React.Component {
               placeholder="Type task name"
               aria-label="Type task name"
               aria-describedby="basic-addon2"
-              onChange={val => this.updateInputValue(val)}
+              onChange={event => this.setInputValue(event.target.value)}
               value={newTaskVal}
             />
+            <InputGroup.Append>
+              <SpeechRecognition resultCallback={this.setInputValue}/>
```

Comment:
Please read an article on variables naming. I can recommend this one: https://ru.hexlet.io/blog/posts/naming-errors-1

From what I see here, I absolutely cannot understand what `resultCallback` can stand for.